### PR TITLE
 removing redundant variables and constraints

### PR DESCRIPTION
**Pull request by aivarsoo**
Minor fixes to the example:

Readme modifications:
1. Comment added on `bins.lowest_num_bin`
1. A width constraint is modified and commented on. Now the text in readme corresponds to the code

Code modifications:
1. Without loss of generality we can assign the first case to the first bin, therefore the variable `bin_loc[0,0]` is fixed to be equal to `1`.
2. Since the minimum number of bins is estimated and saved in `bins.lowest_num_bin`, we can fix the first `bins.lowest_num_bin` `bin_on[j]` variables to be equal to one. Thus we can remove `bins.lowest_num_bin` variables, as well as,  `2 * bins.lowest_num_bin` constraints. 
3. The width constraints are now added only once
4. The length constraint for `j=0` is not add, since it is redundant.
5. All the inequality constraints are changed to have be lower or equal than `<=` for consistency.
6. The order of adding the constraints is modified so that the `add_discrete` are added first.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,60 +116,74 @@ Therefore, it is possible that larger cases are packed over smaller cases withou
...
+ - $r_{i,k}$: binary variable defining $k$ orientations for case $i$
+ - $x_i$, $y_i$, $z_i$: continuous variable defining location of the back lower left corner of case $i$ along $x$, $y$, and $z$ axes of the bin
+
+Note that we can set some variables *before* the optimization. For example, without loss of generality we can assign the first case to 
+the first bin. Therefore $u_{0,0} = 1$ and $u_{0,j} = 0$ for any $j\ne 0$. Further, we can estimate a lower bound on the number of occupied bins:
+
+$$j_{\rm min} = \left \lceil{\frac{\sum\limits_{i\in I} l_i \cdot w_i \cdot h_i}{L \cdot W \cdot H}}\right \rceil$$
+
+Therfore the first $j_{\rm min}$ bin variables $v_j$ can be set equal to $1$ without
+loss of generality
```
Comment (thisac) [inline README.md:157]:
```suggestion
The first $j_{\rm min}$ bin variables $v_j$ can therefore be set equal to $1$ without
loss of generality.
```

File: README.md [hunk 2]
```diff
@@ -179,76 +193,104 @@ The total objective value is the summation of all these terms.
...
 
-When `u_(i,j)` is `0` these constraints are relaxed.
+#### Bin Boundary Constraints:
+These sets of constraints ensure that case $i$ is not placed outside the bins:
+
+$$
+\begin{aligned}
+j L u_{i,j} - x_i &\le 0  && &\forall i \in I,\\
+n L u_{i,j}  + x_i + x_i' &\le  n L + L(j+1) && \forall j \in J&\forall i\in I,\\
+y_i + y_i' &\le W && \forall j \in J&\forall i \in I,\\
```
Comment (thisac) [inline README.md:288]: Was this wrong in the current docs, or in the code?

File: utils.py [hunk 3]
```diff
@@ -240,10 +240,11 @@ def write_solution_to_file(solution_file_path: str,
     """
     num_cases = cases.num_cases
     num_bins = bins.num_bins
+    lowest_num_bin = bins.lowest_num_bin
     dx, dy, dz = effective_dimensions
     if num_bins > 1:
-        num_bin_used = sum([vars.bin_on[j].energy(sample)
-                            for j in range(num_bins)])
+        num_bin_used = lowest_num_bin + sum([vars.bin_on[j].energy(sample)
+                            for j in range(lowest_num_bin, num_bins)])
```
Comment (thisac) [inline utils.py:247]: Was the issue here that the `energy()` method was missing for vars indices less than `lowest_num_bin`? I'm probably missing something here, but why should the number of bins used be the sum of all energies?

File: packing3d.py [hunk 4]
```diff
@@ -241,7 +247,7 @@ def _define_objective(cqm: ConstrainedQuadraticModel, vars: Variables,
 
     # Third term of the objective:
     third_obj_term = quicksum(
-        bins.height * vars.bin_on[j] for j in range(num_bins))
+        bins.height * vars.bin_on[j] for j in range(lowest_num_bin, num_bins))
```
Comment (hbarovertwo) [inline packing3d.py:250]: I'm curious how changing the range lower bound here works. I know `bin_on` is now fixed to `1` for 1 or more bins, but don't we still need a term in the objective for those bins?

**Review by hbarovertwo — COMMENTED** [review]
Tested out everything and all seems to work OK. Really just one minor comment on the objective that isn't clear to me.

**Review by hbarovertwo — APPROVED** [review]
LGTM

**Pull request merged**